src/id.rs

```diff
@@ -252,8 +258,34 @@ impl ViewId {
         self.request_changes(ChangeFlags::LAYOUT)
     }
 
+    fn is_in_active_window(&self) -> bool {
+        if let Some(root) = self.root() {
+            return crate::window_handle::get_current_view() == root;
+        }
+        false
+    }
+
+    fn force_repaint(&self) -> bool {
+        if let Some(window_id) = self.window_id() {
+            // This should not fail unless, say, the view was never
+            // added to a parent view that is realized
+            if crate::window_tracking::force_window_repaint(&window_id) {
+                return true;
+            }
+        }
+        crate::window_tracking::force_all_repaint()
+    }
+
+    pub fn window_id(&self) -> Option<WindowId> {
+        self.root().and_then(window_id_for_root)
+    }
+
     pub fn request_paint(&self) {
+        let active = self.is_in_active_window();
         self.add_update_message(UpdateMessage::RequestPaint);
+        if !active {
+            self.force_repaint();
```

Comment:
do we still need this given that we've now processing all update messages, and if there's a repaint needed, the repaint request would be sent for that window?